src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/ClusterGrainDirectory.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+using Orleans.SystemTargetInterfaces;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    internal class ClusterGrainDirectory : SystemTarget, IClusterGrainDirectory
```

Comment:
Agreed. I have added a comment:

```csharp
    /// This system target provides an entry point for remote clusters to call directory functions.
    /// Since remote clusters do not have ring information about this cluster, they cannot send 
    /// requests directly to the silo with the right directory partition
    /// unless they know the activation address already. Therefore,
    /// This class serves as an intermediary that forwards the requests to the correct silo.
```
Also, thanks for pointing out the inconsistency w.r.t.  `UnregisterAfterNonexistingActivation`. Turns out this should not be there at all anymore  (because it is now called directly on `LocalGrainDirectory` without going through `ClusterGrainDirectory` first). It was dead code, I have removed it.